src/app.py

```diff
@@ -1,11 +1,53 @@
-from flask import Flask
+from flask import Flask, redirect, render_template, request
```

Comment:
Putting some thoughts on project structure here - nothing to do with this file!

* Minor and more personal preference than strong recommendation, but I like having a [poetry.toml](https://github.com/corndeladmin/DevOps-Course-Starter/blob/master/poetry.toml) file so that my virtual environment is created locally (i.e. in a `.venv` folder within the repo) so that I can easily see/find/delete that - otherwise you have to think a little harder to track it down
* We should definitely add some simple set up instructions to your Readme, e.g.:
  * Project requires Poetry, with [link to docs](https://python-poetry.org/docs/#installation) for installation advice
  * Create a `.env` file based on the `.env.template`
  * `poetry install` to set up dependencies
  * `poetry run flask run` to run the project in dev
* When I run `poetry install` I get a warning - it doesn't seem to prevent dependencies actually installing, nor me running the app, but it says: 
  * > "The current project could not be installed: No file/folder found for package playlist-manager".
  * I typically see this when I name the code folder something other than the package name (e.g. if `src` was instead called `playlist_manager` I think this goes away
  * My preference isn't that rename though, but to explicitly tell poetry to use that folder, e.g. by adding the following `packages` section to your `pyproject.toml`:
```toml
[tool.poetry]
name = "playlist-manager"
<...>
readme = "README.md"
packages = [
    { include = "src" }
]
```

Reply:
Thanks Jack, this is all really useful. Gonna write down any questions I have on the above here but no need to answer, I can book a 1-1 later on in the month. 

- Is that all that's needed in the poetry.toml file? I'm struggling to find some good docs on the settings for it.
- The poetry install warning seems interesting. I can't seem to recreate it. But perhaps I've set something up locally by accident. I'll add that in anyway!

Reply:
(feel free to book anyway, but GitHub emailed me so going to reply here!)

* My `poetry.toml` files tend to be very simple; you can extend them further if you want, but no need to until you want a setting
* That sounds very plausible - maybe I have a different Python/Poetry version that's causing the issue! We could have a play with some Dockerfiles and see if we can reproduce

Reply:
Yeah, perhaps let's take a look a bit further down the line. It may be that dockerising everything will either bring out or resolve the issue anyway.